common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerTokenProvider.java

```diff
@@ -0,0 +1,36 @@
+package com.microsoft.identity.common.internal.broker;
+
+import android.app.Activity;
+import android.content.Intent;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
+
+
+public class BrokerTokenProvider {
+
+    private static final String TAG = BrokerTokenProvider.class.getSimpleName();
+
+    /**
+     * Acquire token interactively, will prompt user if possible.
+     * See {@link MicrosoftAuthServiceHandler#getIntentForInteractiveRequest(final Context context)} for details.
+     */
+    public void acquireTokenInteractively(final Activity activity)
+            throws ClientException {
+        final String methodName = ":acquireTokenInteractively";
+
+        if (activity == null) {
+            throw new ClientException(ErrorStrings.EMPTY_ANDROID_CONTEXT);
+        }
+
+        final Intent brokerIntent = MicrosoftAuthServiceHandler.getInstance().getIntentForInteractiveRequest(activity);
```

Comment:
You are missing adding all the bundle params to this intent, please look at https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java#L648
Let me know if it is done somewhere else.